### PR TITLE
Ktlint in cipd

### PR DESCRIPTION
**Pull request by gmackall**
Creates a [ktlint](https://github.com/pinterest/ktlint/) cipd package, for use in https://github.com/flutter/flutter/pull/143478.

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read the [Flutter Style Guide] _recently_, and have followed its advice.
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [ ] I added new tests to check the change I am making, or this PR is [test-exempt].
- [ ] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: CODEOWNERS [hunk 1]
```diff
@@ -41,6 +41,7 @@ cipd_packages/codesign/**                                                   @Xil
 cipd_packages/device_doctor/**                                              @yusuf-goog
 cipd_packages/doxygen/**                                                    @gspencergoog
 cipd_packages/ruby/**                                                       @godofredoc
+cipd_packages/ktlint/**                                                     @gmackall
```
Comment (keyonghan) [inline CODEOWNERS:44]: Nit: order alphabetically.

File: .ci.yaml [hunk 2]
```diff
@@ -155,6 +155,17 @@ targets:
       - cipd_packages/ruby/**
       - .ci.yaml
 
+  - name: Linux ktlint
```
Comment (keyonghan) [inline .ci.yaml:158]: Do you need one for other platforms?

**Comment by keyonghan** [comment]
> Hey @godofredoc , follow up to internal discussion (took me longer to get to this than expected). Does this look roughly correct? Does it matter where the file goes after it is downloaded by `build.sh`? I saw other scripts making a `build` directory, but couldn't tell how they were pushing the output of the build to cipd

The upload happens in recipes (via a bot) to upload the `build` to cipd. https://github.com/flutter/cocoon/pull/3529 to update the doc.

File: .ci.yaml (hunk 2 above)
Comment (gmackall) [reply .ci.yaml:158]: No, I only plan to use this for one of the linux checks

**Review by keyonghan — APPROVED** [review]
LGTM

**Pull request merged**